client/store/products.js

```diff
@@ -34,12 +45,39 @@ export const fetchOneProduct = productId => async dispatch => {
   }
 }
 
+export const addNewProduct = product => async dispatch => {
+  try {
+    const {data} = await axios.post(`api/products`, product)
+    dispatch(addProduct(data))
+  } catch (err) {
+    console.error(err)
+  }
+}
+
+export const editOneProduct = (id, product) => async dispatch => {
+  try {
+    const {data} = await axios.put(`api/products/${id}`, product)
+    dispatch(editProduct(data))
+  } catch (err) {
+    console.error(err)
+  }
+}
+
 export const productsReducer = (state = initialState, action) => {
   switch (action.type) {
     case GET_PRODUCTS:
       return {...state, allProducts: [...action.products]}
     case GET_ONEPRODUCT:
       return {...state, selectedProduct: action.product}
+    case ADD_PRODUCT:
+      return {...state, allProducts: [...state.allProducts, action.product]}
+    case EDIT_PRODUCT:
+      const objIndex = state.allProducts.findIndex(
```

Comment:
I'm wondering (based on my other comments) if we should be updating the selectedProduct within the EDIT_PRODUCT action reducer as well